### DISCUSSION TITLE
ModuleParseError: Module parse failed: Unexpected token

### DISCUSSION THREAD

**Discussion by yashwanth1796** [webpack/webpack#18651] [Q&A]
I have this webpack config for to run my storybook.
I have package that exported from a different app and published to npm when i use that in storybook it fails with below error

i think it's because the compiled code has graphql queries 

Could i get some help here. Thanks

**ERROR**

 [../node_modules/html-webpack-plugin/lib/loader.js!../node_modules/@storybook/core-common/templates/index.ejs] 2.03 KiB {HtmlWebpackPlugin_0} [built]
WARN Force closed manager build
ModuleParseError: Module parse failed: Unexpected token (101:25995)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
| `])));function al(e){var r=Object.assign({},tl,e);return $r.useQuery(nl,r)}function il(e){var r=Object.assign({},tl,e);return $r.useLazyQuery(nl,r)}var ol=function(){var r,t,n,a,o,s=il({variables:{permissions:[],withPermissions:!1}}),u=s[0],l=s[1],c=l.loading,d=l.data;if(w.useEffect(function(){u()},[]),c||!(d!=null&&d.user))return null;var v=d==null?void 0:d.user,p=v.team,m=(r=v.teamMembership)!=null?r:{isAdmin:!1,memberType:""},g=(t=v.package)!=null?t:{},I=g.isAnnual,S=(p==null?void 0:p.ssoState)==="REQUIRED"?(n=v.permissions)==null?void 0:n.includes(G.UsersView):(a=v.permissions)==null?void 0:a.includes(G.UsersInvite),T=(v==null||(o=v.features)==null?void 0:o.powerCasualEnabled)&&m.memberType==="CASUAL";return T?i.jsx(ug,{}):S?i.jsx(mg,{}):!p&&I?i.jsx(vg,{}):p?null:i.jsx(cg,{})},sl={exports:{}};(function(e){function r(t,n){if(t==null)return{};var a={},o=Object.keys(t),s,u;for(u=0;u<o.length;u++)s=o[u],!(n.indexOf(s)>=0)&&(a[s]=t[s]);return a}e.exports=r,e.exports.__esModule=!0,e.exports.default=e.exports})(sl);var pg=sl.exports;const Ar=Tr(pg);var gg=Ee.createMemoStyles(function(e){var r=e.palette,t=e.spacing,n=e.type;return{navButton:function(o){var s=o.active,u=o.isLogo;return{boxSizing:"border-box",fontWeight:n.fontWeight.medium,fontFamily:n.fontFamily.base,fontSize:n.fontSize.bodySm,"-webkit-font-smoothing":"antialiased",display:"flex",justifyContent:"center",alignItems:"center",cursor:"pointer",position:"relative",borderRadius:0,border:"none",boxShadow:"none",background:"none",textDecoration:"none",color:r.text.light,borderBottom:s?"solid 4px #fff":"solid 4px transparent",borderTop:"solid 4px transparent",margin:u?0:[0,t[4]],padding:0,height:48,"&:hover, &:focus":{borderBottom:"solid 4px #fff",textDecoration:"none"}}}}}),hg=["children","active","href","label","onClick","target","isLogo"],qe=function(r){var t=r.children,n=r.active,a=n===void 0?!1:n,o=r.href,s=o===void 0?void 0:o,u=r.label,l=r.onClick,c=l===void 0?void 0:l,d=r.target,v=d===void 0?void 0:d,p=r.isLogo,m=p===void 0?!1:p,g=Ar(r,hg),I=gg({active:a,isLogo:m}),S=i.jsx("button",{type:"button","aria-label":u,className:I.navButton,onClick:c,"data-testid":g["data-testid"],children:t}),T=i.jsx("a",{href:s,className:I.navButton,title:u,target:v,onClick:c,"data-testid":g["data-testid"],children:t});return s?T:S},Eg=B,yg=je,Cg=function(e,r,t){try{return Eg(yg(Object.getOwnPropertyDescriptor(e,r)[t]))}catch{}},Sg=se,bg=function(e){return Sg(e)||e===null},Ig=bg,$g=String,Tg=TypeError,xg=function(e){if(Ig(e))return e;throw new Tg("Can't set "+$g(e)+" as a prototype")},Og=Cg,Mg=se,Ng=ye,_g=xg,Wa=Object.setPrototypeOf||("__proto__"in{}?function(){var e=!1,r={},t;try{t=Og(Object.prototype,"__proto__","set"),t(r,[]),e=r instanceof Array}catch{}return function(a,o){return Ng(a),_g(o),Mg(a)&&(e?t(a,o):a.__proto__=o),a}}():void 0),Ag=Q,Rg=se,Eo=Wa,ul=function(e,r,t){var n,a;return Eo&&Ag(n=r.constructor)&&n!==t&&Rg(a=n.prototype)&&a!==t.prototype&&Eo(e,a),e},jg=oe,ll=function(){var e=jg(this),r="";return e.hasIndices&&(r+="d"),e.global&&(r+="g"),e.ignoreCase&&(r+="i"),e.multiline&&(r+="m"),e.dotAll&&(r+="s"),e.unicode&&(r+="u"),e.unicodeSets&&(r+="v"),e.sticky&&(r+="y"),r},wg=ie,Pg=Ce,Lg=Or,Dg=ll,yo=RegExp.prototype,cl=function(e){var r=e.flags;return r===void 0&&!("flags"in yo)&&!Pg(e,"flags")&&Lg(yo,e)?wg(Dg,e):r},Ya=U,kg=Y,Ka=kg.RegExp,qa=Ya(function(){var e=Ka("a","y");return e.lastIndex=2,e.exec("abcd")!==null}),Ug=qa||Ya(function(){return!Ka("a","y").sticky}),Bg=qa||Ya(function(){var e=Ka("^r","gy");return e.lastIndex=2,e.exec("str")!==null}),za={BROKEN_CARET:Bg,MISSED_STICKY:Ug,UNSUPPORTED_Y:qa},Fg=Oe.f,dl=function(e,r,t){t in e||Fg(e,t,{configurable:!0,get:function(){return r[t]},set:function(n){r[t]=n}})},Hg=Ye,Gg=zt,Vg=X,Wg=ae,Co=Vg("species"),vl=function(e){var r=Hg(e);Wg&&r&&!r[Co]&&Gg(r,Co,{configurable:!0,get:function(){return this}})},Yg=U,Kg=Y,qg=Kg.RegExp,Za=Yg(function(){var e=qg(".","s");return!(e.dotAll&&e.test(`
| `)&&e.flags==="s")}),zg=U,Zg=Y,Qg=Zg.RegExp,fl=zg(function(){var e=Qg("(?<a>b)","g");return e.exec("b").groups.a!=="b"||"b".replace(e,"$<a>c")!=="bc"}),Xg=ae,Qa=Y,zr=B,Jg=Pa,eh=ul,rh=Ke,th=Ga,nh=Aa.f,So=Or,ah=Ju,bo=ue,ih=cl,ml=za,oh=dl,sh=we,uh=U,lh=Ce,ch=Mr.enforce,dh=vl,vh=X,pl=Za,gl=fl,fh=vh("match"),Ve=Qa.RegExp,mr=Ve.prototype,mh=Qa.SyntaxError,ph=zr(mr.exec),Mt=zr("".charAt),Io=zr("".replace),$o=zr("".indexOf),gh=zr("".slice),hh=/^\?<[^\s\d!#%&*+<=>@^][^\s!#%&*+<=>@^]*>/,nr=/a/g,$n=/a/g,Eh=new Ve(nr)!==nr,hl=ml.MISSED_STICKY,yh=ml.UNSUPPORTED_Y,Ch=Xg&&(!Eh||hl||pl||gl||uh(function(){return $n[fh]=!1,Ve(nr)!==nr||Ve($n)===$n||String(Ve(nr,"i"))!=="/a/i"})),Sh=function(e){for(var r=e.length,t=0,n="",a=!1,o;t<=r;t++){if(o=Mt(e,t),o==="\\"){n+=o+Mt(e,++t);continue}!a&&o==="."?n+="[\\s\\S]":(o==="["?a=!0:o==="]"&&(a=!1),n+=o)}return n},bh=function(e){for(var r=e.length,t=0,n="",a=[],o=th(null),s=!1,u=!1,l=0,c="",d;t<=r;t++){if(d=Mt(e,t),d==="\\")d+=Mt(e,++t);else if(d==="]")s=!1;else if(!s)switch(!0){case d==="[":s=!0;break;case d==="(":ph(hh,gh(e,t+1))&&(t+=2,u=!0),n+=d,l++;continue;case(d===">"&&u):if(c===""||lh(o,c))throw new mh("Invalid capture group name");o[c]=!0,a[a.length]=[c,l],u=!1,c="";continue}u?c+=d:n+=d}return[n,a]};if(Jg("RegExp",Ch)){for(var Qe=function(r,t){var n=So(mr,this),a=ah(r),o=t===void 0,s=[],u=r,l,c,d,v,p,m;if(!n&&a&&o&&r.constructor===Qe)return r;if((a||So(mr,r))&&(r=r.source,o&&(t=ih(u))),r=r===void 0?"":bo(r),t=t===void 0?"":bo(t),u=r,pl&&"dotAll"in nr&&(c=!!t&&$o(t,"s")>-1,c&&(t=Io(t,/s/g,""))),l=t,hl&&"sticky"in nr&&(d=!!t&&$o(t,"y")>-1,d&&yh&&(t=Io(t,/y/g,""))),gl&&(v=bh(r),r=v[0],s=v[1]),p=eh(Ve(r,t),n?this:mr,Qe),(c||d||s.length)&&(m=ch(p),c&&(m.dotAll=!0,m.raw=Qe(Sh(r),l)),d&&(m.sticky=!0),s.length&&(m.groups=s)),r!==u)try{rh(p,"source",u===""?"(?:)":u)}catch{}return p},To=nh(Ve),xo=0;To.length>xo;)oh(Qe,Ve,To[xo++]);mr.constructor=Qe,Qe.prototype=mr,sh(Qa,"RegExp",Qe,{constructor:!0})}dh("RegExp");var Ih=ae,$h=Za,Th=Re,xh=zt,Oh=Mr.get,Oo=RegExp.prototype,Mh=TypeError;Ih&&$h&&xh(Oo,"dotAll",{configurable:!0,get:function(){if(this!==Oo){if(Th(this)==="RegExp")return!!Oh(this).dotAll;throw new Mh("Incompatible receiver, RegExp required")}}});var hr=ie,Zt=B,Nh=ue,_h=ll,Ah=za,Rh=xa,jh=Ga,wh=Mr.get,Ph=Za,Lh=fl,Dh=Rh("native-string-replace",String.prototype.replace),Nt=RegExp.prototype.exec,oa=Nt,kh=Zt("".charAt),Uh=Zt("".indexOf),Bh=Zt("".replace),Tn=Zt("".slice),sa=function(){var e=/a/,r=/b*/g;return hr(Nt,e,"a"),hr(Nt,r,"a"),e.lastIndex!==0||r.lastIndex!==0}(),El=Ah.BROKEN_CARET,ua=/()??/.exec("")[1]!==void 0,Fh=sa||ua||El||Ph||Lh;Fh&&(oa=function(r){var t=this,n=wh(t),a=Nh(r),o=n.raw,s,u,l,c,d,v,p;if(o)return o.lastIndex=t.lastIndex,s=hr(oa,o,a),t.lastIndex=o.lastIndex,s;var m=n.groups,g=El&&t.sticky,I=hr(_h,t),S=t.source,T=0,$=a;if(g&&(I=Bh(I,"y",""),Uh(I,"g")===-1&&(I+="g"),$=Tn(a,t.lastIndex),t.lastIndex>0&&(!t.multiline||t.multiline&&kh(a,t.lastIndex-1)!==`

**Webpack config**

`const path = require('path');
const fs = require('fs');
const webpack = require('webpack');

/**
 * Get custom aliases in a package (if config-key exists)
 * @param pkgName package-name to get aliases
 * @returns {Object} aliases
 */
const getPackageCustomAliases = pkgName => {
  const pkg = require(`../../packages/${pkgName}/package.json`);
  const pkgAliases = pkg['smpackages-config']?.aliases ?? {};
  return Object.entries(pkgAliases).reduce(
    (aliases, [aliasKey, aliasPath]) => ({
      ...aliases,
      [aliasKey]: path.resolve(
        aliasPath.replace('<rootDir>', `../packages/${pkgName}`)
      ),
    }),
    {}
  );
};

/**
 * Loop through all packages and get custom aliases.
 * Use `smpackages-config` key in the package.json
 * with a subkey of `aliases` to Opt-In.
 * @returns aliases
 */
const addPackageAliases = () => {
  const pkgList = fs
    .readdirSync(path.resolve(__dirname, '../../packages'), {
      withFileTypes: true,
    })
    .reduce((a, c) => {
      c.isDirectory() && a.push(c.name);
      return a;
    }, []);
  return pkgList.reduce(
    (aliases, name) => ({ ...aliases, ...getPackageCustomAliases(name) }),
    {}
  );
};

module.exports = async ({ config, mode }) => {
  config.node = {
    fs: 'empty',
    tls: 'empty',
    net: 'empty',
    module: 'empty',
    console: true,
    child_process: 'empty',
  };
  config.resolve.extensions = ['.js', '.json', '.ts', '.tsx'];
  config.resolve.modules = [
    'node_modules',
    path.resolve('node_modules'),
    path.resolve('../node_modules'),
  ];

  config.externals = ['jsdom'];
  config.module.rules = config.module.rules.concat([
    {
      test: /\.(js|mjs|ts|tsx)$/,
      exclude: /node_modules/,
      use: [
        {
          loader: require.resolve('babel-loader'),
          options: {
            presets: ['@babel/preset-env', '@babel/preset-react'],
            highlightCode: true,
            rootMode: 'upward',
          },
        },
      ],
    },
    {
      test: /\.(graphql)$/,
      loader: 'graphql-tag/loader',
    },
    {
      test: /\.scss$/,
      loaders: ['style-loader', 'css-loader', 'sass-loader'],
    },
  ]);
  config.plugins.push(
    new webpack.DefinePlugin({
      NODE_ENV: JSON.stringify(process.env.NODE_ENV),
      __DEV__: process.env.NODE_ENV !== 'production',
    })
  );

  return config;
};
`

**Comment by alexander-akait** [comment]
Please create readable reproducible test repo, thank you
